pkg/internal/transform/spanner.go

```diff
@@ -106,6 +108,47 @@ func extractIP(b []uint8, size int) string {
 	return net.IP(b[:size]).String()
 }
 
+func verifyAllHexDigits(hexDigits string) bool {
+	hexBytes, err := hex.DecodeString(hexDigits)
+	if err != nil && hex.DecodedLen(len(hexDigits)) == len(hexBytes) {
+		return true
+	}
+	return false
+}
+
+func extractTraceID(traceparent [55]byte) string {
+	// If traceparent was not set in eBPF, entire field should be zeroed bytes.
+	if traceparent[0] == 0 {
+		return ""
+	}
+
+	// See https://www.w3.org/TR/trace-context/#traceparent-header-field-values for format.
+	// It is assumed that eBPF code has already verified the length is exactly 55
+	// 2 hex version + dash + 32 hex traceID + dash + 16 hex parent + dash + 2 hex flags
+	if !verifyAllHexDigits(string(traceparent[0:2])) {
+		slog.Debug("Traceparent version invalid", "traceparent", string(traceparent[:]))
+		return ""
+	}
+	if traceparent[2] != '-' || traceparent[35] != '-' || traceparent[52] != '-' {
+		slog.Debug("Traceparent missing expected dash", "traceparent", string(traceparent[:]))
+		return ""
+	}
+	traceID := string(traceparent[3:35])
```

Comment:
This is great validation logic, but I wonder if we can live with something much simpler here, something that checks if the length is valid for [3:35], length > 53 and perhaps that it has the two dashes at the expected positions?

I think checking if the values are hex is unnecessary because it's already done in this call:

```
tid, err := trace2.TraceIDFromHex(span.TraceID)
```

Reply:
(duplicate of other conversation, resolving.)